### ISSUE TITLE
DecodeImage中可能存在的错误

### ISSUE THREAD

**Issue by flintning** [PaddlePaddle/PaddleOCR#14698]
### 🔎 Search before asking

- [x] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [x] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [x] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### 🐛 Bug (问题描述)

进行模型微调以识别手写英文字母/符号时，数据集为灰度PNG图片，配置文件定义如下:
Global:
  d2s_train_image_shape: [1, 32, 32]
Train:
  dataset:
    transforms:
    - DecodeImage:
        img_mode: GRAY
        channel_first: false
我一开始以为img_mode是指原始图片的格式，所以把img_mode设置为GRAY，训练过程中一直报错: Invalid number of channels in input image，把GRAY改为BGR，d2s_train_image_shape改为[3, 32, 32]以后，训练正常进行
检查了imgug/operators.py中的DecodeImage算子:
1	class DecodeImage(object):
2	    """decode image"""
3
4	    def __init__(
5	        self, img_mode="RGB", channel_first=False, ignore_orientation=False, **kwargs
6	    ):
7	        self.img_mode = img_mode
8	        self.channel_first = channel_first
9	        self.ignore_orientation = ignore_orientation
10
11	    def __call__(self, data):
12	        img = data["image"]
13	        assert type(img) is bytes and len(img) > 0, "invalid input 'img' in DecodeImage"
14	        img = np.frombuffer(img, dtype="uint8")
15	        if self.ignore_orientation:
16	            img = cv2.imdecode(img, cv2.IMREAD_IGNORE_ORIENTATION | cv2.IMREAD_COLOR)
17	        else:
18	            img = cv2.imdecode(img, 1)
19	        if img is None:
20	            return None
21	        if self.img_mode == "GRAY":
22	            img = cv2.cvtColor(img, cv2.COLOR_GRAY2BGR)
23	        elif self.img_mode == "RGB":
24	            assert img.shape[2] == 3, "invalid shape of image[%s]" % (img.shape)
25	            img = img[:, :, ::-1]
26
27	        if self.channel_first:
28	            img = img.transpose((2, 0, 1))
29
30	        data["image"] = img
31	        return data
第18行，flags为cv2.IMREAD_COLOR，imdecode解码灰度图片返回的img对象总是BGR格式，当mig_mode为GRAY时，第22行转换为BGR时报错
根据代码语义，img_mode应该是指解码后img对象的格式，那么第22行应该是cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)

### 🏃‍♂️ Environment (运行环境)

OS    Windows 11
Python    3.8.20
PaddleOCR main

### 🌰 Minimal Reproducible Example (最小可复现问题的Demo)

Global:
  debug: false
  use_gpu: false
  epoch_num: 100
  log_smooth_window: 20
  print_batch_step: 1
  save_model_dir: ./output/rec_ppocr_v4
  save_epoch_step: 10
  eval_batch_step: [0, 100]
  cal_metric_during_train: true
  pretrained_model: ./train/rec/ch_PP-OCRv4_rec_train/student
  checkpoints:
  save_inference_dir:
  use_visualdl: false
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: ./ppocr/utils/ppocr_keys_v1.txt
  max_text_length: &max_text_length 25
  infer_mode: false
  use_space_char: true
  distributed: true
  save_res_path: ./output/rec/predicts_ppocrv3.txt
  d2s_train_image_shape: [1, 32, 32]

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.0001
    warmup_epoch: 2
  regularizer:
    name: L2
    factor: 3.0e-05

Architecture:
  model_type: rec
  algorithm: SVTR_LCNet
  Transform:
  Backbone:
    name: PPLCNetV3
    scale: 0.95
  Head:
    name: MultiHead
    head_list:
      - CTCHead:
          Neck:
            name: svtr
            dims: 120
            depth: 2
            hidden_dims: 120
            kernel_size: [1, 3]
            use_guide: True
          Head:
            fc_decay: 0.00001
      - NRTRHead:
          nrtr_dim: 384
          max_text_length: *max_text_length

Loss:
  name: MultiLoss
  loss_config_list:
    - CTCLoss:
    - NRTRLoss:

PostProcess:  
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: MultiScaleDataSet
    ds_width: false
    data_dir: D:/Develop/gitee/Paddle/datasets/handwriting1
    ext_op_transform_idx: 1
    label_file_list:
    - D:/Develop/gitee/Paddle/datasets/handwriting1/train_list.txt
    transforms:
    - DecodeImage:
        img_mode: GRAY
        channel_first: false
    - RecConAug:
        #prob: 0.5
        #ext_data_num: 2
        #image_shape: [48, 320, 3]
        #max_text_length: *max_text_length
    - RecAug:
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  sampler:
    name: MultiScaleSampler
    scales: [[28,28],[32,32]]
    first_bs: 200
    fix_bs: false
    divided_factor: [8, 16] # w, h
    is_training: True
  loader:
    shuffle: true
    batch_size_per_card: 200
    drop_last: true
    num_workers: 8
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: D:/Develop/gitee/Paddle/datasets/handwriting1
    label_file_list:
    - D:/Develop/gitee/Paddle/datasets/handwriting1/train_list.txt
    transforms:
    - DecodeImage:
        img_mode: GRAY
        channel_first: false
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - RecResizeImg:
        image_shape: [1, 32, 32]
        padding: True
        keep_ratio: True
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 100
    num_workers: 4

**Comment by Sunting78** [comment]
您好 灰度图目前是会处理为3通道输入的，您使用的网络结构的输入通道固定为3通道，因此灰度图实际也是转为三通道训练的。https://github.com/PaddlePaddle/PaddleOCR/blob/a836921984957d0c63d7799515728f14f089ff85/ppocr/modeling/backbones/rec_lcnetv3.py#L421
